### PR TITLE
Dev platform plugins imt

### PR DESCRIPTION
**Pull request by oleksandrn-imt**

### CODE REVIEW

File: platform/barefoot/sonic-platform-modules-imt/debian/copyright [hunk 1]
```diff
@@ -0,0 +1,13 @@
+This program is free software; you can redistribute it and/or
```
Comment (vitalivanov) [inline platform/barefoot/sonic-platform-modules-imt/debian/copyright:1]: why not our license header?

Comment (oleksandrn-imt) [reply platform/barefoot/sonic-platform-modules-imt/debian/copyright:1]: Not sure we can use our proprietary header in a public repo

**Review by vitalivanov — APPROVED** [review]

File: platform/barefoot/sonic-platform-modules-imt/debian/control [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Source: sonic-platform-modules-imt
+Section: main
+Priority: extra
```
Comment (AndriiKit) [inline platform/barefoot/sonic-platform-modules-imt/debian/control]: As far as I know package priority 'extra' is deprecated.

File: platform/barefoot/sonic-platform-modules-imt/setup.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+setup(
+    name='sonic-platform',
+    version='1.0',
+    description='SONiC platform API implementation',
+    author_email='support@interfacemasters.com',
+    packages=[
+        'sonic_platform',
+        'sonic_platform/pltfm_mgr_rpc',
+    ],
+    package_data = {'sonic_platform':['logging.conf']},
```
Comment (AndriiKit) [inline platform/barefoot/sonic-platform-modules-imt/setup.py]: Seems like we don't have 'logging.conf' file that can be added to our package.

File: platform/barefoot/sonic-platform-modules-imt/sonic_platform/__init__.py [hunk 4]
```diff
@@ -0,0 +1,2 @@
+__all__ = ['chassis', 'psu', 'thermal', 'sensors', 'sfp']
```
Comment (AndriiKit) [inline platform/barefoot/sonic-platform-modules-imt/sonic_platform/__init__.py:1]: Why exactly these modules we want to import with "import *" ?

File: platform/barefoot/sonic-platform-modules-imt/sonic_platform/chassis.py [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+        Retrieves sfp represented by (1-based) index <index>
+
+        Args:
+            index: An integer, the index (1-based) of the sfp to retrieve.
+                   The index should be the sequence of a physical port in a chassis,
+                   starting from 1.
+                   For example, 0 for Ethernet0, 1 for Ethernet4 and so on.
+
+        Returns:
+            An object dervied from SfpBase representing the specified sfp
```
Comment (AndriiKit) [inline platform/barefoot/sonic-platform-modules-imt/sonic_platform/chassis.py]: Just typo - "dervied"

File: platform/barefoot/sonic-platform-modules-imt/sonic_platform/sensors.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+}
+'''
+def _sensors_get() -> dict:
+	data = subprocess.check_output("/usr/bin/sensors -A -u", shell=True, text=True)
+	data = data.split('\n\n')
+	data = [_sensors_chip_parsed(chip_data) for chip_data in data if chip_data]
+	data = dict(data)
+	return data
+
+def _value_get(d: dict, key_prefix, key_suffix=''):
```
Comment (AndriiKit) [inline platform/barefoot/sonic-platform-modules-imt/sonic_platform/sensors.py]: If I'm not mistaken it contradicts the PEP code style that we use function which name started with underscore outside the module

File: platform/barefoot/sonic-platform-modules-imt/sonic_platform/fan_drawer.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    def get_position_in_parent(self):
+        return self.__num
+
+    def is_replaceable(self):
+        return False
+
+    def get_status(self):
+        return True
+
+    # STATUS_LED_COLOR_GREEN = "green"
```
Comment (AndriiKit) [inline platform/barefoot/sonic-platform-modules-imt/sonic_platform/fan_drawer.py]: Don't think we need it

File: platform/barefoot/sonic-platform-modules-imt/sonic_platform/chassis.py [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+    ##############################################
+
+    def get_num_fans(self):
+        """
+        Retrieves the number of fans available on this chassis
+
+        Returns:
+            An integer, the number of fan modules available on this chassis
+        """
+        return len(self._fan_list)
```
Comment (AndriiKit) [inline platform/barefoot/sonic-platform-modules-imt/sonic_platform/chassis.py]: I'm not sure how we work with _fan_list. E.g. for __fan_drawers attribute of Chassis class we have getter property _fan_drawer_list. This getter return list of FanDrawer objects constructed by fan_drawer_list_get public function. But we never did the same for _fan_list. Basically each object of FanDrawer has it's own _fan_list and doesn't look that we reach it here.

File: platform/barefoot/sonic-platform-modules-imt/setup.py (hunk 3 above)
Comment (oleksandrn-imt) [reply platform/barefoot/sonic-platform-modules-imt/setup.py]: fixed

File: platform/barefoot/sonic-platform-modules-imt/debian/control (hunk 2 above)
Comment (oleksandrn-imt) [reply platform/barefoot/sonic-platform-modules-imt/debian/control]: fixed

File: platform/barefoot/sonic-platform-modules-imt/sonic_platform/chassis.py (hunk 5 above)
Comment (oleksandrn-imt) [reply platform/barefoot/sonic-platform-modules-imt/sonic_platform/chassis.py]: fixed

File: platform/barefoot/sonic-platform-modules-imt/sonic_platform/sensors.py (hunk 6 above)
Comment (oleksandrn-imt) [reply platform/barefoot/sonic-platform-modules-imt/sonic_platform/sensors.py]: fixed, also all files checked with pycodestyle and fixed all errors

File: platform/barefoot/sonic-platform-modules-imt/sonic_platform/fan_drawer.py (hunk 7 above)
Comment (oleksandrn-imt) [reply platform/barefoot/sonic-platform-modules-imt/sonic_platform/fan_drawer.py]: fixed

File: platform/barefoot/sonic-platform-modules-imt/sonic_platform/chassis.py (hunk 8 above)
Comment (oleksandrn-imt) [reply platform/barefoot/sonic-platform-modules-imt/sonic_platform/chassis.py]: Yeah, you're right. fixed.

File: platform/barefoot/sonic-platform-modules-imt/sonic_platform/__init__.py (hunk 4 above)
Comment (oleksandrn-imt) [reply platform/barefoot/sonic-platform-modules-imt/sonic_platform/__init__.py:1]: they are used in external demons.

**Review by AndriiKit — APPROVED** [review]

**Pull request merged**